public/js/replay.js

```diff
@@ -229,6 +229,19 @@ function Replay (params) {
   * there are no more flags on either side, disable it.
   */
   this.checkFlagButtons = function(){
+
+    if(this.prev_flag == null || this.flagged.length === 0){
+      $('#flag-left').prop("disabled",true);
+    }else{
+      $('#flag-left').prop("disabled",false);
+    }
+    if(this.next_flag == null || this.flagged.length === 0){
+      $('#flag-right').prop("disabled",true);
+    }
+    else{
+      $('#flag-right').prop("disabled",false);
+    }
+
     $('#flag-left').prop('disabled', this.prev_flag === null);
```

Comment:
We can either remove these lines now, or work the new logic into them, but these lines are doing the same thing as the above if  statements.